mathics/builtin/physchemdata.py

```diff
@@ -123,7 +123,7 @@ def apply_all_properties(self, evaluation):
         return from_python(sorted(_ELEMENT_DATA[0]))
 
     def apply_name(self, name, prop, evaluation):
-        "ElementData[name_?StringQ, prop_]"
+        "ElementData[name_String, prop_]"
```

Comment:
Same thing:

```
 def apply_name(self, name, prop, evaluation):
    "ElementData[name_, prop_]"
    if not isinstance(name, String):
        return
```

Reply:
I like pattern matching, but if performance doesn't, we should obey it.

Reply:
@mmatera I'm thinking, why does pattern matching have a so big overhead? Doesn't it just test against a type and get a specific function for that type? If yes, why the overhead is so big?

Reply:
I repaired that there isn't ``System`String`` in `quick_pattern_test`.

Reply:
> @mmatera I'm thinking, why does pattern matching have a so big overhead? Doesn't it just test against a type and get a specific function for that type? If yes, why the overhead is so big?

It is not specifically pattern matching although that may have its own specific slowness. It is any function call of which pattern matching is one kind. 

This is the kind of thing we need to study and improve. Specific details and data can be found in https://github.com/mathics/Mathics/issues/1568#issuecomment-907909285  We need to study that and try things to fix it.